src/Adapter/CartRule/CommandHandler/EditCartRuleHandler.php

```diff
@@ -64,6 +67,19 @@ public function handle(EditCartRuleCommand $command): void
         }
 
         $this->cartRuleRepository->partialUpdate($cartRule, $updatableProperties);
+
+        if (null !== $command->getAssociatedShopIds()) {
+            // by default table shop association doesn't exist because legacy used to handle it differently,
+            // but it seems to work just fine handling it as all other entities shop associations
+            Shop::addTableAssociation('cart_rule', ['type' => 'shop']);
```

Comment:
As I said in the previous comment, if the cart_rule table was forgotten in the shop tables it should be explicitly added, this feature is too unstable to allow ourselves toggling it on runtime like this IMHO